prime-router/src/main/kotlin/serializers/CsvSerializer.kt

```diff
@@ -53,9 +56,35 @@ class CsvSerializer(val metadata: Metadata) {
         defaultValues: Map<String, String> = emptyMap(),
     ): ReadResult {
         val schema = metadata.findSchema(schemaName) ?: error("Internal Error: invalid schema name '$schemaName'")
-        val rows: List<Map<String, String>> = csvReader().readAllWithHeader(input)
         val errors = mutableListOf<ResultDetail>()
         val warnings = mutableListOf<ResultDetail>()
+        var rows = mutableListOf<Map<String, String>>()
+        csvReader().open(input) {
+            readAllWithHeaderAsSequence().forEach { row: Map<String, String> ->
```

Comment:
Checking number of items is serializer-specific.

So there's a flaw here - someone has to "remember" to implement this check for HL7, etc, as we create new formats.  Better would be to extract this up into an Interface method, to at least act as a reminder of required functionality.